lua/chatcommands/duel.lua

```diff
@@ -0,0 +1,271 @@
+                end
+            end
+            return true
+        end
+    end
+
+    for k, v in pairs(DuelRequests) do
+        if (v.player2 == player2ID and v.points == points) then
+            player1:ChatPrint("[red]Player already has a duel request for that amount.")
+            return true
+        elseif (v.player1 == player1ID) then
+            player1:ChatPrint("[red]You already made a duel request.")
+            return true
+        end
+    end
+
+    for k, v in pairs(AcceptedDuels) do
+        if (v.player2 == player2ID and v.points == points) then
+            player1:ChatPrint("[red]Player already has a duel request for that amount.")
+            return true
+        elseif (v.player1 == player1ID) then
+            player1:ChatPrint("[red]You already made a duel request.")
+            return true
+        end
+    end
+
+    return false
+end
+
+
+function hasAcceptedDuel(ply)
+    for k, v in pairs(ActiveDuels) do
+        if (v.player1 == ply or v.player2 == ply) then
+            return true
+        end
+    end
+
+    for k, v in pairs(AcceptedDuels) do
+        if (v.player1 == ply or v.player2 == ply) then
+            return true
+        end
+    end
+
+    return false
+end
+
+function requestDuel(player1, player2, points, kills)
+    if (not hasPendingDuel(player1, player2, points)) then
+        player1:ChatPrint("[orange]" .. player2:Nick() .. " was challenged to a duel.")
+        player2:ChatPrint("[orange]" .. player1:Nick() .. " challenged you to a duel with a win condition of " .. kills .. " kills and a price of [rainbow]" .. string.Comma(points) .. " points[orange]. Say !duel accept [confirm number of points] to accept.")
+
+
+        table.insert(DuelRequests, {
+            player1 = player1,
+            player2 = player2,
+            points = points,
+            kills = kills,
+            time = CurTime(),
+        })
+    end
+end
+
+timer.Create("Duel", 1, 0, function()
+    NewDuelRequests = {}
+    for k, v in pairs(DuelRequests) do
+        if ((v.time + 30) <= CurTime()) then
+
+            if (v.player1 and v.player2) then
+                v.player1:ChatPrint("[edgy]" .. v.player2:Nick() .. "[orange] doesn't want to fight. Try again later.")
+                v.player2:ChatPrint("[orange]You missed out on a duel with [edgy]" .. v.player1:Nick() .. "[orange].")
+            elseif (not v.player1 and v.player2) then
+                v.player2:ChatPrint("[orange]You missed out on a duel.")
+            elseif (v.player1 and not v.player2) then
+                v.player1:ChatPrint("[red]The player you requested a duel with has left. Try again later.")
+            end
+        else
+            table.insert(NewDuelRequests, v)
+        end
+    end
+
+    DuelRequests = NewDuelRequests
+    local ActiveDuelsNew = {}
+    for k, v in pairs(ActiveDuels) do
+
+        if (v.player1_points >= v.kills) then
+            v.player2:SS_TakePoints(amount)
```

Comment:
What if the losing player spends or transfers their points before the duel is over?

Reply:
Hmm, taking the points at the start of the duel from both sides would work, although it would be shitty if the server crashes or something while the duel is going on.

Reply:
Hmm, taking the points at the start of the duel from both sides would work, although it would be shitty if the server crashes or something while the duel is going on.